[2023-08-08T10:51:47.500+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ETL_Pipeline.transform_data scheduled__2023-08-08T09:00:00+00:00 [queued]>
[2023-08-08T10:51:47.509+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ETL_Pipeline.transform_data scheduled__2023-08-08T09:00:00+00:00 [queued]>
[2023-08-08T10:51:47.509+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2023-08-08T10:51:47.522+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): transform_data> on 2023-08-08 09:00:00+00:00
[2023-08-08T10:51:47.526+0000] {standard_task_runner.py:57} INFO - Started process 61 to run task
[2023-08-08T10:51:47.529+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'ETL_Pipeline', 'transform_data', 'scheduled__2023-08-08T09:00:00+00:00', '--job-id', '5039', '--raw', '--subdir', 'DAGS_FOLDER/my_first_dag.py', '--cfg-path', '/tmp/tmpqj5lee3e']
[2023-08-08T10:51:47.531+0000] {standard_task_runner.py:85} INFO - Job 5039: Subtask transform_data
[2023-08-08T10:51:47.573+0000] {task_command.py:410} INFO - Running <TaskInstance: ETL_Pipeline.transform_data scheduled__2023-08-08T09:00:00+00:00 [running]> on host 48849dae508a
[2023-08-08T10:51:47.683+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='ETL_Pipeline' AIRFLOW_CTX_TASK_ID='transform_data' AIRFLOW_CTX_EXECUTION_DATE='2023-08-08T09:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-08-08T09:00:00+00:00'
[2023-08-08T10:51:47.688+0000] {logging_mixin.py:150} WARNING - Scraping Progress:   0%|          | 0/10 [00:00<?, ?it/s]
[2023-08-08T10:51:54.720+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  10%|#         | 1/10 [00:07<01:03,  7.03s/it]
[2023-08-08T10:52:28.325+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  20%|##        | 2/10 [00:40<03:01, 22.66s/it]
[2023-08-08T10:52:32.360+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  30%|###       | 3/10 [00:44<01:39, 14.16s/it]
[2023-08-08T10:52:33.589+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  40%|####      | 4/10 [00:45<00:54,  9.05s/it]
[2023-08-08T10:52:41.873+0000] {logging_mixin.py:150} WARNING - Scraping Progress:  40%|####      | 4/10 [00:54<01:21, 13.55s/it]
[2023-08-08T10:52:41.873+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/my_first_dag.py", line 40, in transform_data
    data_list.append(scrape_data_from_url(url))
  File "/opt/airflow/dags/utils/scraper.py", line 8, in scrape_data_from_url
    article.parse()
  File "/home/airflow/.local/lib/python3.7/site-packages/newspaper/article.py", line 191, in parse
    self.throw_if_not_downloaded_verbose()
  File "/home/airflow/.local/lib/python3.7/site-packages/newspaper/article.py", line 532, in throw_if_not_downloaded_verbose
    (self.download_exception_msg, self.url))
newspaper.article.ArticleException: Article `download()` failed with HTTPSConnectionPool(host='www.prothomalo.com', port=443): Read timed out. on URL https://www.prothomalo.com/sports/cricket/arnova38c0
[2023-08-08T10:52:41.881+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=ETL_Pipeline, task_id=transform_data, execution_date=20230808T090000, start_date=20230808T105147, end_date=20230808T105241
[2023-08-08T10:52:41.894+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 5039 for task transform_data (Article `download()` failed with HTTPSConnectionPool(host='www.prothomalo.com', port=443): Read timed out. on URL https://www.prothomalo.com/sports/cricket/arnova38c0; 61)
[2023-08-08T10:52:41.927+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-08-08T10:52:41.964+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
